apis/database/v1alpha1/redshift_types.go

```diff
@@ -0,0 +1,652 @@
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="STATE",type="string",JSONPath=".status.atProvider.clusterStatus"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster,categories={crossplane,managed,aws}
+type Cluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ClusterSpec   `json:"spec"`
+	Status ClusterStatus `json:"status,omitempty"`
+}
+
+// ClusterSpec defines the desired state of an AWS Redshift Cluster.
+type ClusterSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  ClusterParameters `json:"forProvider"`
+}
+
+// ClusterParameters define the parameters available for an AWS Redshift cluster
+type ClusterParameters struct {
+	// AllowVersionUpgrade indicates that major engine upgrades are applied automatically to the
+	// cluster during the maintenance window.
+	// default=true
+	// +optional
+	AllowVersionUpgrade *bool `json:"allowVersionUpgrade,omitempty"`
+
+	// AutomatedSnapshotRetentionPeriod is the number of days for which
+	// automated backups are retained. Setting this parameter to a positive
+	// number enables backups. Setting this parameter to  0 disables automated backups.
+	// default=1
+	// +kubebuilder:validation:Maximum=35
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	AutomatedSnapshotRetentionPeriod *int64 `json:"automatedSnapshotRetentionPeriod,omitempty"`
+
+	// AvailabilityZone is the EC2 Availability Zone in which you want
+	// Amazon Redshift to provision the cluster.
+	// Default: A random, system-chosen Availability Zone in the region that is
+	// specified by the endpoint.
+	// Example: us-east-2d
+	// Constraint: The specified Availability Zone must be in the same region as
+	// the current endpoint. The Availability Zone parameter can't be specified
+	// if the MultiAZ parameter is set to true.
+	// The specified Availability Zone must be in the same AWS Region as the current endpoint.
+	// +optional
+	AvailabilityZone *string `json:"availabilityZone,omitempty"`
+
+	// ClusterParameterGroupName is the name of the cluster parameter group to use for the cluster.
+	// Default: The default Amazon Redshift cluster parameter group. For information
+	// about the default parameter group, go to Working with Amazon Redshift Parameter
+	// Groups (https://docs.aws.amazon.com/redshift/latest/mgmt/working-with-parameter-groups.html)
+	// +optional
+	ClusterParameterGroupName *string `json:"clusterParameterGroupName,omitempty"`
+
+	// SecurityGroups is a list of security groups to associate with this cluster.
+	// Default: The default cluster security group for Amazon Redshift.
+	// +optional
+	ClusterSecurityGroups []string `json:"clusterSecurityGroups,omitempty"`
```

Comment:
Can we add referencers for https://doc.crds.dev/github.com/crossplane/provider-aws/ec2.aws.crossplane.io/SecurityGroup/v1beta1 type here?

Reply:
This has been added.